brainiak/utils/fmrisim.py

```diff
@@ -959,77 +1223,129 @@ def calc_noise(volume,
     else:
         trs = list(range(0, volume.shape[3]))
 
-    # Go through the trs and pull out the fwhm
+    # Go through the trs and pull out the fwhm and snr
     fwhm = [0] * len(trs)
     for tr in list(range(0, len(trs))):
         fwhm[tr] = _calc_fwhm(volume[:, :, :, trs[tr]],
-                              mask[:, :, :, trs[tr]],
+                              mask,
                               noise_dict['voxel_size'],
                               )
 
     # Keep only the mean
     noise_dict['fwhm'] = np.mean(fwhm)
-
-    # Calculate the autoregressive and drift noise
-    auto_reg_sigma, drift_sigma = _calc_temporal_noise(volume, mask)
-
-    # Calibrate for how sigma is originally calculated
-    auto_reg_sigma = auto_reg_sigma / noise_dict['temporal_noise']
+    noise_dict['snr'] = _calc_snr(volume,
+                                  mask,
+                                  )
 
     # Total temporal noise, since these values only make sense relatively
-    total_temporal_noise = auto_reg_sigma + drift_sigma
+    total_temporal_noise = auto_reg + drift
 
     # What proportion of noise is accounted for by these variables?
-    noise_dict['auto_reg_sigma'] = auto_reg_sigma / total_temporal_noise
-    noise_dict['drift_sigma'] = drift_sigma / total_temporal_noise
+    noise_dict['auto_reg_sigma'] = auto_reg / total_temporal_noise
+    noise_dict['drift_sigma'] = drift / total_temporal_noise
 
     # Return the noise dictionary
     return noise_dict
 
 
 def _generate_noise_system(dimensions_tr,
+                           spatial_sd,
+                           temporal_sd,
+                           spatial_noise_type='exponential',
```

Comment:
I don't have a citation but based on my experience, the exponential is the best fit. THere is some evidence that at low SNR, gaussian models are the best fit, at higher SNR rician noise is and then at even higher still (what we have with modern methods) I posit that an exponential function is better but I have no proof or theoretical basis, just empirical support

Reply:
I see. By best fit in your experience, do you mean the best fit of the noise outside of the brain?
I am mainly concerned with the issue that exponential distribution is one-sided. Have you considered Laplacian distribution (https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.random.laplace.html)? It is the same as exponential distribution on the positive side but is symmetric (i.e., has mass on the negative side)

Reply:
If you look at a histogram of voxel values in fMRI data what you will see (as long as the SNR is high) is a bimodal distribution where the first mode is at 0 with an exponential drop off. If I used a symmetric distribution I would still need the mode to be 0 and then I would need to round all values below zero to 0 (since the MR signal can't be negative). Hence I would not get an appropriate distribution using laplacian

Reply:
I see. That makes sense.